cpp/serve/engine_actions/action_commons.cc

```diff
@@ -50,8 +64,25 @@ void ProcessFinishedRequestStateEntries(std::vector<RequestStateEntry> finished_
       // So we mark the parent entry as finished.
       rstate->entries[parent_idx]->status = RequestStateStatus::kFinished;
       // Remove the request state entry from all the models.
-      RemoveRequestFromModel(estate, rstate->entries[parent_idx]->mstates[0]->internal_id, models);
-      estate->id_manager.RecycleId(rstate->entries[parent_idx]->mstates[0]->internal_id);
+      if (estate->prefix_cache->HasSequence(rstate->entries[parent_idx]->mstates[0]->internal_id)) {
```

Comment:
This is very similar to the changes in line 32, it's better extracting to a function or lambda

Reply:
making it a function with name